subs/pantry/src/Pantry/Archive.hs

```diff
@@ -387,41 +393,60 @@ parseArchive pli archive fp = do
             case Map.lookup (seSource se) blobs of
               Nothing -> error $ "Impossible: blob not found for: " ++ seSource se
               Just blobKey -> pure (sfp, TreeEntry blobKey (seType se))
-
           -- parse the cabal file and ensure it has the right name
-          (cabalPath, cabalEntry@(TreeEntry cabalBlobKey _)) <- findCabalFile pli tree
+          (cabalPath, cabalEntry@(TreeEntry cabalBlobKey _), btype) <- findCabalOrHpackFile pli tree
```

Comment:
Minor nitpicking but what does "b" in `btype` mean?

Reply:
I have renamed it to  `buildFileType` now.  Hopefully that's better.